trie.rb

```diff
@@ -1,10 +1,52 @@
 require_relative "tree"
 
-# Implement a Trie (sometimes called a "prefix tree")
-# See: http://en.wikipedia.org/wiki/Trie
-#
-# A Trie allows for efficient prefix searching, i.e.,
-# "Find me all things that start with 'Foo'"
-
 class Trie
+  attr_reader :root, :prefix
```

Comment:
Does this work?  The code is kind of hard to follow and there are parts of it that are suspect to me.